### PR TITLE
Bump @azure/communication-calling to new stable version

### PR DESCRIPTION
**Pull request by prprabhu-ms**
# What

* Bump `@azure/communication-calling` to version `1.3.2`
* Make necessary changes to use of some breaking changes in API since last beta
* Make necessary API changes in our own API as a result.

### CODE REVIEW

**Review by JamesBurnside — APPROVED** [review]

**Review by anjulgarg — APPROVED** [review]

File: packages/calling-stateful-client/src/CallAgentDeclarative.test.ts [hunk 1]
```diff
@@ -80,10 +77,9 @@ class MockCallAgent implements CallAgent {
     return call;
   }
   join(groupLocator: GroupLocator, options?: JoinCallOptions): Call;
-  join(groupChatCallLocator: GroupChatCallLocator, options?: JoinCallOptions): Call;
-  join(meetingLocator: MeetingLocator, options?: JoinCallOptions): Call;
+  join(TeamsMeetingLinkLocator: TeamsMeetingLinkLocator, options?: JoinCallOptions): Call;
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  join(meetingLocator: any, options?: any): Call {
+  join(TeamsMeetingLinkLocator: any, options?: any): Call {
```
Comment (PorterNan) [inline packages/calling-stateful-client/src/CallAgentDeclarative.test.ts:82]: Upper case? Lower case? (For first 'T'), this one with any is just implementation, we shouldn't expose it or we should keep it as meetingLocator

**Comment by JamesBurnside** [comment]
Weird, seems to have impacted the static html apps:
image

File: packages/calling-stateful-client/src/CallAgentDeclarative.test.ts [hunk 2]
```diff
@@ -132,18 +125,14 @@ class MockCallAgentWithMultipleCalls implements CallAgent {
...
```
Comment (PorterNan) [inline packages/calling-stateful-client/src/CallAgentDeclarative.test.ts:130]: Same here this shouldn't be renamed

File: packages/calling-stateful-client/src/CallAgentDeclarative.test.ts [hunk 3]
```diff
@@ -132,18 +125,14 @@ class MockCallAgentWithMultipleCalls implements CallAgent {
     return call;
   }
   join(groupLocator: GroupLocator, options?: JoinCallOptions): Call;
-  join(groupChatCallLocator: GroupChatCallLocator, options?: JoinCallOptions): Call;
-  join(meetingLocator: MeetingLocator, options?: JoinCallOptions): Call;
+  join(TeamsMeetingLinkLocator: TeamsMeetingLinkLocator, options?: JoinCallOptions): Call;
```
Comment (PorterNan) [inline packages/calling-stateful-client/src/CallAgentDeclarative.test.ts:128]: teamsMeetingLocator

**Review by PorterNan — APPROVED** [review]
One minor comment about name, since let's make it before ship the npm package

**Comment by PorterNan** [comment]
Just noticed those are bot api file, so we are good to have this checked in, then open a bug item for it

File: packages/calling-stateful-client/src/CallAgentDeclarative.test.ts [hunk 4]
```diff
@@ -80,8 +77,7 @@ class MockCallAgent implements CallAgent {
     return call;
   }
   join(groupLocator: GroupLocator, options?: JoinCallOptions): Call;
-  join(groupChatCallLocator: GroupChatCallLocator, options?: JoinCallOptions): Call;
-  join(meetingLocator: MeetingLocator, options?: JoinCallOptions): Call;
+  join(meetingLocator: TeamsMeetingLinkLocator, options?: JoinCallOptions): Call;
```
Comment (PorterNan) [inline packages/calling-stateful-client/src/CallAgentDeclarative.test.ts:80]: I guess this one can be called teamsMeetingLocator? Should be fine cause this is just test

Comment (PorterNan) [reply packages/calling-stateful-client/src/CallAgentDeclarative.test.ts:80]: And the one typed with any, will be just meetingLocator

Comment (prprabhu-ms) [reply packages/calling-stateful-client/src/CallAgentDeclarative.test.ts:80]: Keeping changes minimal.

**Pull request merged**